book/src/template_expansion.md

```diff
@@ -0,0 +1,387 @@
+        _did_loop = true;
+        writer
+            .write_fmt(
+                format_args!(
+                    "\n    {0}\n",
+                    &::askama::MarkupDisplay::new_unsafe(&(user), ::askama::Html),
+                ),
+            )?;
+    }
+    if !_did_loop {}
+}
+::askama::Result::Ok(())
+```
+
+Interestingly enough, there is a `_did_loop` variable. It is used in case your
+`for` loop has a `if` condition:
+
+```jinja
+{% for user in x if x.len() > 2 %}
+    {{ user }}
+{% else %}
+    {{ x }}
+{% endfor %}
+```
+
+Which generates:
+
+```rust
+{
+    let mut _did_loop = false;
+    let _iter = (&self.x).into_iter();
+    let _iter = _iter.filter(|user| -> bool { self.x.len() > 2 });
+    for (user, _loop_item) in ::askama::helpers::TemplateLoop::new(_iter) {
+        _did_loop = true;
+        writer
+            .write_fmt(
+                format_args!(
+                    "\n    {0}\n",
+                    &::askama::MarkupDisplay::new_unsafe(&(user), ::askama::Html),
+                ),
+            )?;
+    }
+    if !_did_loop {
+        writer
+            .write_fmt(
+                format_args!(
+                    "\n    {0}\n",
+                    &::askama::MarkupDisplay::new_unsafe(
+                        &(self.x),
+                        ::askama::Html,
+                    ),
+                ),
+            )?;
+    }
+}
+::askama::Result::Ok(())
+```
+
+So in case the code doesn't enter in the loop because of the `if` condition,
+it will run the `else` block.
+
+## Filters
+
+Example by using the `abs` built-in filter:
+
+```jinja
+{{ -2|abs }}
+```
+
+Which generates:
+
+```rust
+writer
+    .write_fmt(
+        format_args!(
+            "{0}",
+            &::askama::MarkupDisplay::new_unsafe(
+                &(::askama::filters::abs(-2)?),
+                ::askama::Html,
+            ),
+        ),
+    )?;
+::askama::Result::Ok(())
+```
+
+The filter is called with `-2` as first argument. If you add arguments to the
+call, like this:
+
+```jinja
+{{ "a"|indent(4) }}
+```
+
+Which generates:
+
+```rust
+writer
+    .write_fmt(
+        format_args!(
+            "{0}",
+            &::askama::MarkupDisplay::new_unsafe(
+                &(::askama::filters::indent("a", 4)?),
+                ::askama::Html,
+            ),
+        ),
+    )?;
+::askama::Result::Ok(())
+```
+
+No surprise there, `4` is added after `"a"`. Now let's check when we chain the filters:
+
+```jinja
+{{ "a"|indent(4)|capitalize }}
+```
+
+Which generates:
+
+```rust
+writer
+    .write_fmt(
+        format_args!(
+            "{0}",
+            &::askama::MarkupDisplay::new_unsafe(
+                &(::askama::filters::capitalize(
+                    &(::askama::filters::indent("a", 4)?),
+                )?),
+                ::askama::Html,
+            ),
+        ),
+    )?;
+::askama::Result::Ok(())
+```
+
+As expected, `capitalize` first argument is the value returned by the `indent` call.
```

Comment:
This is missing a possessive something, maybe "`capitalize`'s"?

Reply:
Good catch!